### PR TITLE
build: generate OpenAPI from grpc

### PR DESCRIPTION
**Pull request by erka**
Use google/gnostic to generate OpenAPI v3 schema base on the proto files.

This PR has mixed `option (google.api.http)` and `grpc_api_configuration=rpc/flipt/flipt.yaml` together. If the path should be included into OpenAPI spec it should be use `google.api.http` for path configuration

Issues:
- [x] what is the best place for `openapi.yaml`
- [x] how to check impact of replacement `github.com/google/gnostic/openapiv3` with `github.com/google/gnostic-models/openapiv3`
- [x] OpenAPI security settings probably doesn't have all options
- [x] ~legacy evaluation endpoints are missing. Is it a problem?~
- [x] should OFREP endpoints be part of Flipt OpenAPI?

related flipt-io/flipt-openapi/issues/25

### CODE REVIEW

**Comment by markphelps** [comment]
Looks great! just replying to some of your questions in the PR description:

> what is the best place for openapi.yaml

I think at the root of the project? We could setup an action maybe to copy it over to https://github.com/flipt-io/flipt-openapi which drives the updates in our docs, or maybe just remove `flipt-openapi` altogether and update the docs action to pull the file from here?

It seems that Mintlify can pull the openapi spec from anywhere, so maybe we can give it a link to the raw file on GitHub at the root?

CleanShot 2024-08-06 at 15 43 53

> legacy evaluation endpoints are missing. Is it a problem?

No. We don't want to show the legacy evaluation endpoints.

> should OFREP endpoints be part of Flipt OpenAPI?

Yes. I think we should add this as part of the OpenAPI docs

**Review by GeorgeMac — APPROVED** [review]
I think this is great @erka 🙌 Awesome catch. Yeah, I think as Mark suggested, maybe get it into the root of the project? Then lets get it in 💪

**Comment by erka** [comment]
@markphelps could I rename `operation_id` for OpenApi?

**Comment by markphelps** [comment]
> @markphelps could I rename `operation_id` for OpenApi?

@erka how do you mean? do you mean for OFREP? or something else?

File: rpc/flipt/evaluation/evaluation.proto [hunk 1]
```diff
@@ -232,7 +233,7 @@ message EvaluationNamespaceSnapshotRequest {
   string reference = 2;
 }
 
-// flipt:sdk:ignore
 service DataService {
+  option (google.api.api_visibility) = {restriction: "internal"};
```
Comment (markphelps) [inline rpc/flipt/evaluation/evaluation.proto:237]: we should probably make this public as far as OpenAPI docs go. although we still don't want to generate the SDK methods for it (GRPC  / REST)

**Comment by markphelps** [comment]
> > how do you mean? do you mean for OFREP? or something else?
> 
> I played a bit with other side of OpenApi generation and I would like to make the api a bit nicer there. For example, we have such operation_ids currently
> 
> ```yaml
>  /api/v1/namespaces:
>         get:
>             tags:
>                 - Flipt
>             operationId: NamespacesService.list
> --- omitted ---
> /ofrep/v1/evaluate/flags/{key}:
>         post:
>             tags:
>                 - OFREPService
>             operationId: ofrep.evaluate
> ```
> 
> After code generation, I get the client with the methods
> 
> ```go
> func main() {
> 	client, _ := gen.NewClientWithResponses("http://localhost:8080")
> 
> 	client.NamespacesServiceList(context.TODO(), ...)
> 
> 	client.OfrepEvaluate(context.TODO(), "flag", ...)
> }
> ```
> 
> It isn't very clear for me what `NamespacesServiceList` does and I prefer `GetNamespaces` method there but that may be a breaking changes for someone who uses current OpenApi spec.
> 
> @markphelps wdyt?

I think that's fine, although I would suggest `ListNamespaces`

**Review by markphelps — APPROVED** [review]
lgtm! thank you @erka !!

**Pull request merged**